### DISCUSSION TITLE
Item permissions - compare two fields

### DISCUSSION THREAD

**Discussion by jeremybradbury** [directus/directus#11334] [Feature Requests]
AFAIK you can only compare a field to a static value, this seems like a UX / UI only limitation, as adding a second value on the other side of the equals sign seems fairly simple. 

For example, here I tried to add that a user who was assigned or created a task could edit the description, edit if it was done or not or reassign it.

For example, I make you a task, we can both edit all the things, but everyone else can only see it.

However, I was only able to compare these 2 variables to static input and not each other. 

![item-permissions](https://user-images.githubusercontent.com/1664387/151649807-d51fb774-499f-4244-aae6-ca2078731537.PNG)

FWIW: I recently went down a similar rabbit hole with Strapi and it became some big issue wrapped up in the user & permissions tooling. I think there is a really easy fix in Directus' case, but Strapi still needs a lot more work to accomplish the same goals.

Not that it matters too much: I'm a fluffy bunny rabbit trainer, so I really dig the lagomorph vibe around here.

What probably matters more: if there's a fix that just need some work, point me in it's general direction and I'll get on it. I only need a tiny bit of context.

**Comment by licitdev** [comment]
I believe what you need is [Dynamic Variables](https://docs.directus.io/configuration/filter-rules/#dynamic-variables).
You may use an `OR` group to allow either the creator or assignee in the update permissions.

![Screenshot 2022-01-31 at 1 32 58 PM](https://user-images.githubusercontent.com/26413686/151744026-ea431ef6-c9ec-49bd-93f1-768e0fef2176.png)

**Reply by jeremybradbury** [reply]
@licitdev TYVM, this is nifty and works for my use case.

my idea was the same exact picker on the other side of the equals.

if we're asking SQL (and/or app code) to compare a database field to a static item, it's not much more complex to compare 2 fields on the same or related records, and needs less documentation than the current "dynamic variables" solution.

**Reply by licitdev** [reply]
>  compare 2 fields on the same or related records

Could it be something along the lines of a new dynamic variable called `$CURRENT_ITEM.something`? 
However, I think it may get trickier as there are different "depths" of an item when relationships are involved.

**Reply by jeremybradbury** [reply]
Well if I wanted to do something simple like ensure those params are equal on the same record assigned and created by, I guess.

What if we had an item picker (like m2a) so we could say is this user, this user or this user (or any other collection)?

I think the dynamic variable pattern is not the best pattern for other uses (besides current user).

In SQL you can put any 2 fields on either side of a filter. Not just is this field like/equals/less than this static number or string or date. Current user is app injected data not SQL fields.

If we add a collection & item picker on the right of the equals, then default to the same object on the left and allow us to pick another object, it would fix / extend filtering to be far more useful than just current user or static options... Like manually entered UUIDs.

All the while still not replacing the usefulness of the current user dynamic variable solution.

**Reply by licitdev** [reply]
> In SQL you can put any 2 fields on either side of a filter.

Yes, the comparison between two columns could be pretty nifty. 👍🏻 Thank you for the clarification!

On a side note, the implementation may not be as simple as we will have to deal with raw queries for comparisons between columns. It could get tricky when multiple table joins are involved (from relational fields).

**Comment by jeremybradbury** [comment]
Yeah, I personally always use PostgreSQL, but I understand that you don't,
and for good reasons.

This problem is usually solved with a subquery to find an ID or field to
match, so something like...

```sql
SELECT * FROM directus_users where id = (SELECT id FROM directus_users
where wallet_id = 7).id
```

In this case we're checking that we match a user id from a static user's
wallet id, which is a relationship to a 1:1 table to users where users link
their wallets.

See how we avoided a join by knowing the PK of the related field we want to
compare?

To me this subquery isn't any more complex than the original query, even if
we still needed joins for both queries.

AND since we're currently storing the static data to compare to... Like a
string based UUID...

How hard is it to do the look up for the item data once, in the UI and then
store its ID (or other raw value) in the same place that you would put
manually entered text?

There seem to be several ways to solve this, with a wide range of LOE.

As of now I'm having to go look up records in the database and copy and
paste their uuid into a static field.

It should be much simpler.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
